File: src/LinkedList/DoubleLinkedList/DoubleNode.java
package LinkedList.DoubleLinkedList;

public class DoubleNode<T> {
	private DoubleNode<T> next;
	private DoubleNode<T> previous;
	private T data;
	
	public DoubleNode(){
		
	}
	public DoubleNode(T data,DoubleNode<T> previous,DoubleNode<T> next){
		this.data = data;
		this.next = next;
		this.previous = previous;
	}
	
	public boolean isNIL(){
		return this.data == null;
	}
	
	
	public T getData(){
		return this.data;
	}
	
	public DoubleNode<T> getNext(){
		return this.next;
	}
	public DoubleNode<T> getPrevious(){
		return this.previous;
	}
	
	public void setData(T element){
		this.data = data;
	}
	public void setNext(DoubleNode<T> next){
		this.next = next;
	}
	
	public void setPrevious(DoubleNode<T> previous){
		this.previous = previous;
	}
}

File: Python/LinkedList/DoubleLinkedList/DoubleNode.py


class DoubleNode():
	
	def __init__(self,data=None,previous=None,nextNode=None):
		self.__data = data
		self.__previous = previous
		self.__nextNode = nextNode
	
	
	def isEmpty(self):
		return (self.__data == None)
	
	
	def getData(self):
		return self.__data
	
	
	def setData(self,newData):
		self.__data = newData

	
	def getNext(self):
		return self.__nextNode
	
	
	def setNext(self,newNext):
		self.__nextNode = newNext
	
	
	def getPrevious(self):
		return self.__previous
	
	
	def setPrevious(self,newPrevious):
		self.__previous = newPrevious
	
	
		
		
